net/keepalived/files/etc/init.d/keepalived-inotify

```diff
@@ -0,0 +1,42 @@
+#!/bin/sh /etc/rc.common
+
+START=99
+USE_PROCD=1
+PROG="/usr/bin/keepalived-rsync-inotify.sh"
+
+start_keepalived_inotify_instance() {
+	local cfg=$1
+
+	config_get name $cfg name
+	[ -z "$name" ] && return
+
+	config_get sync $cfg sync 0
+	[ "$sync" = "0" ] && return
+
+	config_get sync_mode $cfg sync_mode
+	[ "$sync_mode" != "receive" ] && return
+
+	config_get sync_dir $cfg sync_dir /usr/share/keepalived/rsync
+	[ ! -d "$sync_dir" ] && mkdir -p "$sync_dir"
+
+	procd_open_instance "$name"
+	procd_set_param command /bin/sh "$PROG" "$name" "$sync_dir"
+	procd_set_param pidfile /var/run/keepalived-inotify-$name.pid
+	procd_close_instance
+}
+
+stop_keepalived_inotify_instance() {
+	local cfg=$1
+
+	killall inotifywait
```

Comment:
I don't think that's so good idea.
You'll kill all the other scripts which use inotifywait!
Do you even need it? Doesn't the procd do that for you?

Reply:
> I don't think that's so good idea.
> You'll kill all the other scripts which use inotifywait!

handled with the signal handler and fifo.

>  Do you even need it? Doesn't the procd do that for you?

I am not sure if it's possible to trigger hotplug event with it or not. 

This service keeps watch on the rsync directory for added/deleted/updated files and then it triggers hotplug event.

In the keepalived hotplug event it passes an extra argument `inotify` and variables
```sh
ACTION=SYNC
RSYNC_SOURCE=<source_file> # location of the file on the local host received from the master node
RSYNC_TARGET=<actual_file> # actual location of the file when this file should be replaced
```

If `/etc/config/dhcp` is changed on the master node, rsync.sh script will rsync that file to a backup node on configured interval.

On backup node, keepalived-rsync-inotify service, will get a modify event and it would trigger
```sh
ACTION=SYNC RSYNC_SOURCE="/usr/share/keepalived/rsync/etc/config/dhcp" \
 RSYNC_TARGET="/etc/config/dhcp" \
 /sbin/hotplug-call keepalived inotify
```

Whenever the master/backup switchover happens, to avoid delay, this hotplug event can be used to copy files to the target location and execute any action like start/stop service or execute any other command.